R/add_quantiles.R

```diff
@@ -0,0 +1,154 @@
+#' @title Add lines indicating the survival quantiles to a survival plot
+#'
+#' @description Method to add lines indicating the specified quantiles of the survival curve.
+#'
+#' @param gg visR object
+#' @param ... other arguments passed on to the method to modify \code{\link[ggplot2]{geom_line}}
+#'
+#' @examples
+#' 
+#' library(visR)
+#' 
+#' adtte %>%
+#'   estimate_KM("SEX") %>%
+#'   visr() %>%
+#'   add_quantiles() 
+#' 
+#' adtte %>%
+#'   estimate_KM("SEX") %>%
+#'   visr() %>%
+#'   add_quantiles(c(0.25, 0.50, 0.75)) 
+#' 
+#' adtte %>%
+#'   estimate_KM("SEX") %>%
+#'   visr() %>%
+#'   add_quantiles(c(0.25, 0.50, 0.75)) 
+#'
+#' @return Lines indicating the quantiles overlayed on a visR ggplot
+#'
+#' @rdname add_quantiles
+#'
+#' @export
+
+add_quantiles <- function(gg, ...){
+  UseMethod("add_quantiles", gg)
+}
+
+#' @param gg A ggplot created with visR
+#' @param quantiles y-axis values for which to draw the lines (default: 0.5)
+#' @param linetype aesthetic of ggplot2 \code{\link[ggplot2]{geom_line}} (default: dashed)
+#' @param linecolour aesthetic of ggplot2 \code{\link[ggplot2]{geom_line}} (default: black)
+#' @param alpha aesthetic of ggplot2 \code{\link[ggplot2]{geom_line}} (default: 0.8)
+#' @param ... other arguments passed on to the method to modify \code{\link[ggplot2]{geom_line}}
+#'
+#' @rdname add_quantiles
+#' @method add_quantiles ggsurvfit
+#' @export
+
+add_quantiles.ggsurvfit <- function(gg, 
+                                    quantiles = 0.5,
+                                    linetype = "mixed",
+                                    linecolour = "strata",
+                                    alpha = 1, 
+                                    ...) {
+  
+  if (!is.numeric(quantiles)) {
+    
+    stop("Invalid argument for `quantiles`, must be numeric. Setting it to 0.5.")
+    quantiles <- 0.5
+    
+  }
+  
+  if ((alpha > 1) | (alpha < 0)) {
+    
+    warning("Invalid `alpha` argument, must be between 0 and 1. Setting it to 1.0.")
+    alpha <- 1.0
+    
+  }
+  
+  model <- eval(gg$data$call[[1]])
```

Comment:
we will get errors if only strata if we dont add the overall strata manually see add_risktable

Reply:
fixed, good catch

Reply:
I dont see the fix